src/Health/HealthController.ts

```diff
@@ -98,6 +107,13 @@ export class HealthController {
     return pickWalletInfoKeys(await this.bitcoinCore.getWalletInfo())
   }
 
+  private async checkWalletBalance(walletInfo: WalletInfo): Promise<WalletInfo> {
```

Comment:
This looks like a pure function that could be unit tested. Name could be better too, try to think through what this functions true responsibility is.

Reply:
Ok I updated the name and moved the checking logic to another function with unit tests. Looking better?